internal/db/metamodel.go

```diff
@@ -20,77 +20,269 @@ var ModelModel = Model{
-}
-
-var ModelRelationshipsLeft = Relationship{
-	Id:           uuid.MustParse("806334bf-98ce-4c08-87f4-5d9bed4f6d60"),
-	LeftModelId:  uuid.MustParse("872f8c55-9c12-43d1-b3f6-f7a02d937314"), // model
-	LeftName:     "leftRelationships",
-	LeftBinding:  HasMany,
-	RightModelId: uuid.MustParse("90be6901-60a0-4eca-893e-232dc57b0bc1"), // relationship
-	RightName:    "leftModel",
-	RightBinding: BelongsTo,
-}
-
-var ModelRelationshipsRight = Relationship{
-	Id:           uuid.MustParse("3ccee9ea-f5b7-4707-9b0f-9e6f96ddd42e"),
-	LeftModelId:  uuid.MustParse("872f8c55-9c12-43d1-b3f6-f7a02d937314"), // model
-	LeftName:     "rightRelationships",
-	LeftBinding:  HasMany,
-	RightModelId: uuid.MustParse("90be6901-60a0-4eca-893e-232dc57b0bc1"), // relationship
-	RightName:    "rightModel",
-	RightBinding: BelongsTo,
-}
-
 var AttributeModel = Model{
-	Id:   uuid.MustParse("14d840f5-344f-4e23-af12-d4caa1ffa848"),
+	ID:   uuid.MustParse("14d840f5-344f-4e23-af12-d4caa1ffa848"),
 	Name: "attribute",
 	Attributes: map[string]Attribute{
 		"name": Attribute{
-			Id:       uuid.MustParse("51605ada-5326-4cfd-9f31-f10bc4dfbf03"),
-			AttrType: String,
-		},
-		"attrType": Attribute{
-			Id:       uuid.MustParse("c29a6558-7676-40a8-be00-e0933342efd7"),
-			AttrType: Enum,
+			ID:       uuid.MustParse("51605ada-5326-4cfd-9f31-f10bc4dfbf03"),
+			Datatype: String,
 		},
 	},
 	RightRelationships: []Relationship{
 		ModelAttributes,
 	},
+	LeftRelationships: []Relationship{
+		AttributeDatatype,
+	},
 }
 
 var RelationshipModel = Model{
-	Id:   uuid.MustParse("90be6901-60a0-4eca-893e-232dc57b0bc1"),
+	ID:   uuid.MustParse("90be6901-60a0-4eca-893e-232dc57b0bc1"),
 	Name: "relationship",
 	Attributes: map[string]Attribute{
 		"leftName": Attribute{
-			Id:       uuid.MustParse("3e649bba-b5ab-4ee2-a4ef-3da0eed541da"),
-			AttrType: String,
+			ID:       uuid.MustParse("3e649bba-b5ab-4ee2-a4ef-3da0eed541da"),
+			Datatype: String,
 		},
 		"rightName": Attribute{
-			Id:       uuid.MustParse("8d8524ab-92d6-49a3-8038-3ad957c5f6e8"),
-			AttrType: String,
+			ID:       uuid.MustParse("8d8524ab-92d6-49a3-8038-3ad957c5f6e8"),
+			Datatype: String,
 		},
 		"leftBinding": Attribute{
-			Id:       uuid.MustParse("3c0b2893-a074-4fd7-931e-9a0e45956b08"),
-			AttrType: Int,
+			ID:       uuid.MustParse("3c0b2893-a074-4fd7-931e-9a0e45956b08"),
+			Datatype: Int,
 		},
 		"rightBinding": Attribute{
-			Id:       uuid.MustParse("4135be16-7c61-4750-b53d-f1eeff69086e"),
-			AttrType: Int,
+			ID:       uuid.MustParse("4135be16-7c61-4750-b53d-f1eeff69086e"),
+			Datatype: Int,
 		},
 	},
 	RightRelationships: []Relationship{
 		ModelRelationshipsRight,
 		ModelRelationshipsLeft,
 	},
 }
+
+var DatatypeModel = Model{
+	ID:   uuid.MustParse("c2ea9d6f-26ca-4674-b2b4-3a2bc3861a6a"),
+	Name: "datatype",
+	Attributes: map[string]Attribute{
+		"name": Attribute{
+			ID:       uuid.MustParse("0a0fe2bc-7443-4111-8b49-9fe41f186261"),
+			Datatype: String,
+		},
+		"storageFormat": Attribute{
```

Comment:
maybe prefer storageType to "format" which suggests like.. string formatting to me

Reply:
I debated one way or another on this. Open to going back to type